tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -460,5 +497,169 @@ private static bool ParseBooleanConfigurationValue(string value)
+            return false;
+        }
+
+        private static void NoManageCodeClrVersionIis(ProcessInfo process)
+        {
+            foreach (var module in process.Modules)
+            {
+                var fileName = Path.GetFileName(module);
+
+                if (fileName.Equals("Microsoft.AspNetCore.Server.IIS.dll", StringComparison.OrdinalIgnoreCase))
+                {
+                    AnsiConsole.WriteLine(TracingDotnetOnIis);
+                }
+            }
+        }
+
+        private static string? CheckWindowsInstallation()
+        {
+            string? tracerVersion = null;
+            string? displayName = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                const string datadog64BitProgram = "Datadog .NET Tracer 64-bit";
+                const string datadog32BitProgram = "Datadog .NET Tracer 32-bit";
+
+                var registrySubKeys = new[] { @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall", @"SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall" };
+
+                foreach (var registryKey in registrySubKeys)
+                {
+                    using (var key = Registry.LocalMachine.OpenSubKey(registryKey))
+                    {
+                        if (key is not null)
+                        {
+                            foreach (var subKeyName in key.GetSubKeyNames())
+                            {
+                                var subKey = key.OpenSubKey(subKeyName);
+
+                                if (subKey?.GetValue("DisplayName") is datadog64BitProgram or datadog32BitProgram && displayName is null)
+                                {
+                                    displayName = (string?)subKey.GetValue("DisplayName");
+                                    tracerVersion = subKey.GetValue("VersionMajor") + "." + subKey.GetValue("VersionMinor");
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (displayName is null)
+                {
+                    Utils.WriteError(TraceProgramNotFound);
+                }
+                else if (displayName is datadog64BitProgram)
+                {
+                    Utils.WriteSuccess(TracerProgramFound(datadog64BitProgram));
+                }
+                else if (displayName is datadog32BitProgram)
+                {
+                    if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                    {
+                        Utils.WriteError(WrongTracerArchitecture(datadog32BitProgram));
+                        return tracerVersion;
+                    }
+
+                    Utils.WriteSuccess(TracerProgramFound(datadog32BitProgram));
+                }
+            }
+
+            return tracerVersion;
+        }
+
+        private static void CheckAppPools(ApplicationPool? pool)
```

Comment:
I'd rather keep this in CheckIisCommand:
 - There's nothing related to the worker process here
 - As much as possible, I'd like to keep IIS objects (here, ApplicationPool) from spreading in other parts of the code

Reply:
Thank you for the confirmation! I'll go ahead and move stuff around since that makes sense.

Reply:
Hey @kevingosse, please let me know if the current changes look good, seems this can't be merge until your confirmation! 🙇